include/uxr/agent/scheduler/FCFSScheduler.hpp

```diff
@@ -45,12 +45,14 @@ class FCFSScheduler : public Scheduler<T>
 template<class T>
 inline void FCFSScheduler<T>::init()
 {
+    std::lock_guard<std::mutex> lock(mtx_);
     running_cond_ = true;
```

Comment:
`running_cond_` could be standard bool type. Not required to be atomic as `mtx_` protects it.

Reply:
Fixed!